### DISCUSSION TITLE
Two questions to ask

### DISCUSSION THREAD

**Discussion by vmtest888** [flet-dev/flet#2398] [Q&A] [answered]
Sorry, I have two questions to ask.   I don't have much understanding of Flet, maybe the problem itself is wrong

1.  Is it  possible for Python to quickly translate through tools and directly use the existing ecosystem on Flutter pub.dev

2.  Are you considering using embedded Python and using Flutter to directly load Python's dynamic libraries? This way, there can be only one process

**Comment by mjrodri** [comment, marked as answer]
While it's technically possible to use Python in conjunction with Flutter, the integration might not be as straightforward as using Flutter's native Dart language. Flutter primarily uses Dart to build its UI components, and while there are ways to run Python code within a Flutter app, it typically involves additional layers of complexity.

One approach to using Python in a Flutter app is through the use of platform channels, where you can communicate between Dart (Flutter's language) and a native codebase (such as Python) using message passing. However, this might introduce challenges related to performance, maintainability, and compatibility.

Using embedded Python and loading its dynamic libraries directly in Flutter could be an option, but it requires careful consideration of the implications. You would need to ensure that the Python libraries are compatible with the Flutter environment, and managing the interaction between Dart and Python could be non-trivial.

Before pursuing this approach, it's essential to evaluate whether it's the most suitable solution for your specific use case. Additionally, be aware that embedding a scripting language like Python within a mobile app might impact performance and increase the app's size.

If you're looking to leverage existing Python code within a Flutter app, consider alternatives like creating a REST API for your Python functionality and interacting with it through HTTP requests from your Flutter app. This can provide a more modular and maintainable architecture.

Always weigh the pros and cons of different integration approaches based on your project requirements and constraints.
